content/src/components/editor/components/share-button.tsx

```diff
@@ -58,15 +72,42 @@ function ShareContent() {
   }, [copied, setCopied])
   const handleCopyLink = useCallback(() => {
     if (Result.isSuccess(result)) {
-      navigator.clipboard.writeText(result.value)
+      navigator.clipboard.writeText(result.value.url)
       setCopied(true)
     }
   }, [setCopied, result])
 
+  const [downloaded, setDownloaded] = useState(false)
+
+  useEffect(() => {
+    if (!downloaded) return
+    const timeout = setTimeout(() => {
+      setDownloaded(false)
+    }, 2000)
+    return () => {
+      clearTimeout(timeout)
+    }
+  }, [downloaded, setDownloaded])
+  const handleDownloadLink = useCallback(() => {
+    if (Result.isSuccess(result)) {
+      var blobUrl = URL.createObjectURL(result.value.zipFile.content)
+
+      var link = document.createElement("a")
+      link.href = blobUrl
+      link.download = result.value.zipFile.name
+      link.innerText = "Click here to download the file"
+      document.body.appendChild(link)
+      link.click()
+      setDownloaded(true)
+    }
+  }, [setDownloaded, result])
+
   return (
     <Fragment>
       <div className="flex flex-col space-y-2 text-center sm:text-left">
-        <h3 className="text-lg text-[--sl-color-white] font-semibold">Share</h3>
+        <h3 className="text-lg text-[--sl-color-white] font-semibold">
```

Comment:
perhaps someone can tell me why the recommended prettier plugin is not formatting correctly.